### PR TITLE
stop using g_jit outside of JitInterface

### PR DESCRIPTION
**Pull request by CrystalGamma**
Replace g_jit in x86-64 ASM routines code by m_jit member reference.
Followup on #7599.

### CODE REVIEW

**Comment by JonnyH** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.h [hunk 1]
```diff
@@ -19,10 +19,13 @@ namespace MMIO
 class Mapping;
 }
 
+class Jitx86Base;
+
 // Like XCodeBlock but has some utilities for memory access.
 class EmuCodeBlock : public Gen::X64CodeBlock
 {
 public:
+  EmuCodeBlock(Jitx86Base& jit) : m_jit{jit} {}
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.h]: `explicit EmuCodeBlock(...`

File: Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.h [hunk 2]
```diff
@@ -24,6 +25,7 @@ class QuantizedMemoryRoutines : public EmuCodeBlock
 class CommonAsmRoutines : public CommonAsmRoutinesBase, public QuantizedMemoryRoutines
 {
 public:
+  CommonAsmRoutines(Jitx86Base& jit) : QuantizedMemoryRoutines(jit) {}
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.h]: `explicit CommonAsmRoutines(...`

File: Source/Core/Core/PowerPC/Jit64Common/Jit64Base.h [hunk 3]
```diff
@@ -32,14 +32,15 @@ constexpr size_t CODE_SIZE = 1024 * 1024 * 32;
 
 class Jitx86Base : public JitBase, public QuantizedMemoryRoutines
 {
+public:
+  Jitx86Base() : QuantizedMemoryRoutines(*this), trampolines(*this) {}
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64Base.h]: `trampolines` can be initialized like how `blocks` is.

File: Source/Core/Core/PowerPC/Jit64Common/TrampolineCache.h [hunk 4]
```diff
@@ -24,6 +24,7 @@ class TrampolineCache : public EmuCodeBlock
   const u8* GenerateWriteTrampoline(const TrampolineInfo& info);
 
 public:
+  TrampolineCache(Jitx86Base& jit) : EmuCodeBlock(jit) {}
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64Common/TrampolineCache.h]: `explicit TrampolineCache(...`

File: Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.h [hunk 5]
```diff
@@ -13,6 +13,7 @@ enum EQuantizeType : u32;
 class QuantizedMemoryRoutines : public EmuCodeBlock
 {
 public:
+  QuantizedMemoryRoutines(Jitx86Base& jit) : EmuCodeBlock(jit) {}
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.h]: `explicit QuantizedMemoryRoutines(...`

**Comment by lioncash** [comment]
> (I didn't even know about that feature … how long has it been in the language?).

Since C++98

> I hope there are no objections on a more general level?

Nope

**Comment by CrystalGamma** [comment]
Hmm, before looking at Dolphin I only ever looked at pre-standard versions of C++, specifically MSVC6 … shows how much I know :S

**Comment by JonnyH** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/PowerPC/Jit64Common/Jit64Base.h [hunk 6]
```diff
@@ -32,14 +32,15 @@ constexpr size_t CODE_SIZE = 1024 * 1024 * 32;
...
-
 public:
+  Jitx86Base() : QuantizedMemoryRoutines(*this) {}
   JitBlockCache* GetBlockCache() override { return &blocks; }
   bool HandleFault(uintptr_t access_address, SContext* ctx) override;
+
+protected:
+  bool BackPatch(u32 emAddress, SContext* ctx);
+  JitBlockCache blocks{*this};
+   TrampolineCache trampolines{*this};
```
Comment (CrystalGamma) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64Base.h]:
```suggestion
  TrampolineCache trampolines{*this};
```

**Pull request merged**